xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -326,17 +326,13 @@ func routesProtoToSlice(routes []*v3routepb.Route, csps map[string]clusterspecif
 					}
 					wc.HTTPFilterConfigOverride = cfgs
 					route.WeightedClusters[c.GetName()] = wc
+
+					// Sum of all weights cannot exceed MaxUint32. Checking for overflow.
+					if totalWeight > (totalWeight + w) {
```

Comment:
What if we declared `totalWeight` as a `uint64` instead? Then we can unconditionally add `w` to `totalWeight` and simply check if `totalWeight > math.MaxUint32`. I feel that will be more readable. What do you think?

Reply:
I agree. Updating.